packages/plugin/src/rules/require-import-fragment.ts

```diff
@@ -0,0 +1,116 @@
+import { FragmentSpreadNode } from 'graphql';
+import { GraphQLESTreeNode } from '../estree-converter/index.js';
+import { GraphQLESLintRule } from '../types.js';
+
+const RULE_ID = 'require-import-fragment';
+const SUGGESTION_ID = 'add-import-expression';
+
+export const rule: GraphQLESLintRule = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      category: 'Operations',
+      description: 'Require fragments to be imported via an import expression.',
+      url: `https://github.com/B2o5T/graphql-eslint/blob/master/docs/rules/${RULE_ID}.md`,
+      examples: [
+        {
+          title: 'Incorrect',
+          code: /* GraphQL */ `
+            query MyQuery {
+              fooField {
+                ...Foo
+              }
+            }
+          `,
+        },
+        {
+          title: 'Incorrect',
+          code: /* GraphQL */ `
+            # import Bar from 'bar.graphql'
+            # import 'foo.graphql'
+
+            query MyQuery {
+              fooField {
+                ...Foo
+              }
+            }
+          `,
+        },
+        {
+          title: 'Correct',
+          code: /* GraphQL */ `
+            # import Foo from 'foo.graphql'
+
+            query MyQuery {
+              fooField {
+                ...Foo
+              }
+            }
+          `,
+        },
+      ],
+    },
+    hasSuggestions: true,
+    messages: {
+      [RULE_ID]: "Expected '{{name}}' fragment to be imported.",
+      [SUGGESTION_ID]: "Add import expression for '{{name}}'",
+    },
+    schema: [],
+  },
+  create(context) {
+    const knownFragmentNames = new Set<string>();
+    const fragmentSpreadNodes = new Set<GraphQLESTreeNode<FragmentSpreadNode>>();
```

Comment:
```suggestion
    const fragmentSpreadNameNodes = new Set<GraphQLESTreeNode<NameNode>>();
```
we can simplify to push just fragment NameNode